cargo install error on Alpine -- nix related?

When running `cargo install ironbar` on my machine, I got this error:

```
   Compiling ironbar v0.15.1
error[E0432]: unresolved import `nix::fcntl::OFlag`
 --> /home/brian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ironbar-0.15.1/src/clients/wayland/wlr_data_control/offer.rs:3:5
  |
3 | use nix::fcntl::OFlag;
  |     ^^^^^^^^^^^^^^^^^ no `OFlag` in `fcntl`

error[E0432]: unresolved import `nix::unistd::pipe2`
    --> /home/brian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ironbar-0.15.1/src/clients/wayland/wlr_data_control/offer.rs:4:26
     |
4    | use nix::unistd::{close, pipe2};
     |                          ^^^^^
     |                          |
     |                          no `pipe2` in `unistd`
     |                          help: a similar name exists in the module: `pipe`
     |
note: found an item that was configured out
    --> /home/brian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/nix-0.27.1/src/unistd.rs:1233:8
     |
1233 | pub fn pipe2(flags: OFlag) -> Result<(RawFd, RawFd)> {
     |        ^^^^^
     = note: the item is gated behind the `fs` feature

error[E0432]: unresolved imports `nix::fcntl::fcntl`, `nix::fcntl::F_GETPIPE_SZ`, `nix::fcntl::F_SETPIPE_SZ`
   --> /home/brian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ironbar-0.15.1/src/clients/wayland/wlr_data_control/mod.rs:13:18
    |
13  | use nix::fcntl::{fcntl, F_GETPIPE_SZ, F_SETPIPE_SZ};
    |                  ^^^^^  ^^^^^^^^^^^^  ^^^^^^^^^^^^ no `F_SETPIPE_SZ` in `fcntl`
    |                  |      |
    |                  |      no `F_GETPIPE_SZ` in `fcntl`
    |                  no `fcntl` in `fcntl`
    |
    = help: consider importing this module instead:
            nix::fcntl
note: found an item that was configured out
   --> /home/brian/.cargo/registry/src/index.crates.io-6f17d22bba15001f/nix-0.27.1/src/fcntl.rs:509:8
    |
509 | pub fn fcntl(fd: RawFd, arg: FcntlArg) -> Result<c_int> {
    |        ^^^^^
    = note: the item is gated behind the `fs` feature

For more information about this error, try `rustc --explain E0432`.
error: could not compile `ironbar` (bin "ironbar") due to 3 previous errors
error: failed to compile `ironbar v0.15.1`, intermediate artifacts can be found at `/tmp/cargo-installgmhXjZ`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

I then proceeded to clone the repo and build directly (`cargo build --release`), and it worked! I am using Alpine Linux, so not sure if there is some weirdness going on there.  But just wanted to bring it to your attention.  Looking forward to playing with the bar.

**System information:**
 - Distro: Alpine Linux Edge
- Compositor: labwc
 - Ironbar version: 0.15.1

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -125,7 +125,7 @@ mlua = { version = "0.9.7", optional = true, features = ["luajit"] }
 cairo-rs = { version = "0.18.5", optional = true, features = ["png"] }
 
 # clipboard
-nix = { version = "0.27.1", optional = true, features = ["event"] }
+nix = { version = "0.28.0", optional = true, features = ["event", "fs"] }
 
 # clock
 chrono = { version = "0.4.38", optional = true, default_features = false, features = ["clock", "unstable-locales"] }

--- a/src/clients/wayland/wlr_data_control/mod.rs
+++ b/src/clients/wayland/wlr_data_control/mod.rs
@@ -11,7 +11,7 @@ use crate::{lock, try_send, Ironbar};
 use device::DataControlDevice;
 use glib::Bytes;
 use nix::fcntl::{fcntl, F_GETPIPE_SZ, F_SETPIPE_SZ};
-use nix::sys::epoll::{Epoll, EpollCreateFlags, EpollEvent, EpollFlags};
+use nix::sys::epoll::{Epoll, EpollCreateFlags, EpollEvent, EpollFlags, EpollTimeout};
 use smithay_client_toolkit::data_device_manager::WritePipe;
 use smithay_client_toolkit::reexports::calloop::{PostAction, RegistrationToken};
 use std::cmp::min;
@@ -274,7 +274,7 @@ impl DataControlDeviceHandler for Environment {
                     Ok(token) => {
                         cur_offer.token.replace(token);
                     }
-                    Err(err) => error!("{err:?}"),
+                    Err(err) => error!("Failed to insert read pipe event: {err:?}"),
                 }
             }
         }
@@ -349,11 +349,12 @@ impl DataControlSourceHandler for Environment {
                     .add(fd, epoll_event)
                     .expect("to send valid epoll operation");
 
+                let timeout = EpollTimeout::from(100u16);
                 while !bytes.is_empty() {
                     let chunk = &bytes[..min(pipe_size as usize, bytes.len())];
 
                     epoll_fd
-                        .wait(&mut events, 100)
+                        .wait(&mut events, timeout)
                         .expect("Failed to wait to epoll");
 
                     match file.write(chunk) {

--- a/src/clients/wayland/wlr_data_control/offer.rs
+++ b/src/clients/wayland/wlr_data_control/offer.rs
@@ -5,7 +5,7 @@ use nix::unistd::{close, pipe2};
 use smithay_client_toolkit::data_device_manager::data_offer::DataOfferError;
 use smithay_client_toolkit::data_device_manager::ReadPipe;
 use std::ops::DerefMut;
-use std::os::fd::{BorrowedFd, FromRawFd};
+use std::os::fd::{AsFd, AsRawFd};
 use std::sync::{Arc, Mutex};
 use tracing::{trace, warn};
 use wayland_client::{Connection, Dispatch, Proxy, QueueHandle};
@@ -176,11 +176,11 @@ pub unsafe fn receive(
     // create a pipe
     let (readfd, writefd) = pipe2(OFlag::O_CLOEXEC)?;
 
-    offer.receive(mime_type, BorrowedFd::borrow_raw(writefd));
+    offer.receive(mime_type, writefd.as_fd());
 
-    if let Err(err) = close(writefd) {
+    if let Err(err) = close(writefd.as_raw_fd()) {
         warn!("Failed to close write pipe: {}", err);
     }
 
-    Ok(FromRawFd::from_raw_fd(readfd))
+    Ok(ReadPipe::from(readfd))
 }

